packages/fxa-auth-server/lib/payments/google-play/subscriptions.ts

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Container from 'typedi';
+import { internalValidationError } from '../../../lib/error';
+import { AppConfig } from '../../types';
+import { PlayBilling } from './play-billing';
+import { AbbrevPlayPurchase } from 'fxa-shared/subscriptions/types';
+import { SubscriptionPurchase } from './subscription-purchase';
+
+// TODO move this when we add support for Apple IAP subscriptions
+export interface SubscriptionsService<T> {
+  getSubscriptions: (uid: string) => Promise<T[]>;
+}
+
+/**
+ * Extract an AbbrevPlayPurchase from a SubscriptionPurchase
+ */
+export function abbrevPlayPurchaseFromSubscriptionPurchase(
+  purchase: SubscriptionPurchase
+): AbbrevPlayPurchase {
+  return {
+    auto_renewing: purchase.autoRenewing,
+    expiry_time_millis: purchase.expiryTimeMillis,
+    package_name: purchase.packageName,
+    sku: purchase.sku,
+    ...(purchase.cancelReason && { cancel_reason: purchase.cancelReason }),
+  };
+}
+
+export class PlaySubscriptions
+  implements SubscriptionsService<AbbrevPlayPurchase>
+{
+  private playBilling?: PlayBilling;
+
+  constructor() {
+    const config = Container.get(AppConfig);
+    if (!config.subscriptions.enabled) {
+      throw internalValidationError(
+        'PlaySubscriptions',
+        {},
+        new Error(
+          'Trying to new up PlaySubscriptions while subscriptions are disabled.  Check your dependency graph.'
+        )
+      );
+    }
+
+    if (Container.has(PlayBilling)) {
+      this.playBilling = Container.get(PlayBilling);
+    }
+  }
+
+  async getSubscriptions(uid: string) {
```

Comment:
The main change is that I moved the Play subscriptions fetching function here.

Reply:
> 
> 
> This method name is a little misleading IMO ...

That's fair.  That name is from https://github.com/mozilla/fxa/pull/10948/files/dae26867f7073d09f3cb612059cf52c5aa165c91#diff-902713579f413cf56c1200a47bf8dfa9e80d3de15bb44ef04196904e032834d6R13-R15  That's because I want the same interface for any subscription type.  (I suppose we could argue whether that's helpful when the returned objects could be wildly different.)  If you change the name to something Play specific, then we cannot use that interface for web or Apple subscriptions.  And that's fine, but let's remove that interface in your PR.

Reply:
(What the ... why is my reply here?!)